packages/fury-adapter-oas3-parser/lib/parser/oas/parseComponentsObject.js

```diff
@@ -156,9 +156,20 @@ function parseComponentsObject(context, element) {
 
       object.forEach((value, key) => {
         if (value) {
-          // eslint-disable-next-line no-param-reassign
-          value.meta.id = key.clone();
-          array.push(value);
+          if (value instanceof namespace.elements.AuthScheme) {
+            // eslint-disable-next-line no-param-reassign
+            value.meta.id = key.clone();
+            array.push(value);
+
+            return;
+          }
+
+          // append oauth2 flow names
+          value.forEach((flow) => {
+            // eslint-disable-next-line no-param-reassign
+            flow.meta.id = `${key.toValue()} ${flow.grantType}`;
```

Comment:
Context has a method `registerId()` which can be used to register IDs and know if they are unique. If there is a schema with the `id` which could clash we can figure it out and error out.

Reply:
Moved to https://github.com/apiaryio/api-elements.js/issues/330